src/validator_tests.zig

```diff
@@ -0,0 +1,261 @@
+const libssz = @import("lib.zig");
+const utils = libssz.utils;
+const serialize = libssz.serialize;
+const deserialize = libssz.deserialize;
+const hashTreeRoot = libssz.hashTreeRoot;
+const std = @import("std");
+const ArrayList = std.ArrayList;
+const expect = std.testing.expect;
+
+// Beacon chain Validator struct for compatibility testing
+const Validator = struct {
+    pubkey: [48]u8,
+    withdrawal_credentials: [32]u8,
+    effective_balance: u64,
+    slashed: bool,
+    activation_eligibility_epoch: u64,
+    activation_epoch: u64,
+    exit_epoch: u64,
+    withdrawable_epoch: u64,
+};
+
+test "Validator struct serialization" {
```

Comment:
move this to test.zig, not required with the validator list tests already being there, rename this file as list_tests and add bit/byte list test here

Reply:
so actually we have that already in the tests.zig, just move the test "Zeam-style List/Bitlist usage with tree root stability" to here and add expected bytes comparision using fastssz as well

Reply:
no let the slices test be there untouched

Reply:
just rename this file as beacon_tests.zig